### PR TITLE
[10.0][ADD] business code and validation

### PR DESCRIPTION
**Pull request by jarmokortetjarvi**
Adds a business id for partners.
Split to two modules: one just adds the field and other adds validation and formatting.
This way the validation will be optional, and the module can be installed in environments with incorrectly formatted business ids

### CODE REVIEW

**Review by aisopuro — APPROVED** [review]
My concerns on the previous version of this pull request (#15) were already addressed. This is OK.

File: README.md
Comment (mlaitinen) [inline README.md]: Maybe add `l10n_fi_business_code` and `l10n_fi_business_code_validate` here too.

Comment (mlaitinen) [reply README.md]: I'll take that back. This section is autogenerated: https://raw.githubusercontent.com/OCA/maintainer-quality-tools/master/sample_files/README.md

File: setup/_metapackage/VERSION.txt
Comment (mlaitinen) [inline setup/_metapackage/VERSION.txt]: Is there a reason why all this `setup/` stuff is in here? Isn't this something that OCA Bot generates automatically?

File: README.md
Comment (mlaitinen) [reply README.md]: Maybe remove this modification (edicode) and let OCA bot autogenerate it.

**Review by mlaitinen — CHANGES_REQUESTED** [review]

File: README.md
Comment (jarmokortetjarvi) [reply README.md]: Done. The auto-generation was the reason why I didn't add the business code -modules in readme.

File: setup/_metapackage/VERSION.txt
Comment (jarmokortetjarvi) [reply setup/_metapackage/VERSION.txt]: As far as I know, they are autogenerated for https://pypi.org/project/setuptools-odoo/
They seem to be here due to a merge https://github.com/OCA/l10n-finland/pull/16/commits/46277158a44c9f01f0bdc4e469dfd2a716a5f479 from main branch

Comment (mlaitinen) [reply setup/_metapackage/VERSION.txt]: Ok. Could you remove them from this PR?

Comment (mlaitinen) [reply setup/_metapackage/VERSION.txt]: @aisopuro Do you know how we could get rid of those commits? I tried to remove them by using interactive rebase but failed miserably.

Comment (aisopuro) [reply setup/_metapackage/VERSION.txt]: It looks a little hairy. There are some merge commits in there, which could make the whole thing really complicated. Maybe try a reset-"rebase":
1. `git checkout -b just-testing 10.0-l10n_fi_business_code`
2. `git fetch origin`
3. `git reset --soft origin/10.0` (Assuming `origin` is the remote pointing to the OCA repository)
4. `git add the-things-you-want-here` (make sure you don't add any `setup/` stuff if it isn't wanted)
5. `git commit`
5. (check the branch makes sense, then `reset --hard` your feature branch to point to this fixed branch)
6. `git push --force-with-lease`

This should "squash" everything so that you only have one commit with the files you want to have. You can also do steps 4 and 5 several times, if you want to have different commits cover different things.

**Comment by jarmokortetjarvi** [comment]
Tried to clean up the commit history. If everything went as planned, it should be identical to the original PR (with fixes of course), but without any `./setup`-files.

Pre-cleanup branch for reference https://github.com/Tawasta/l10n-finland/tree/10.0-l10n_fi_business_code_original

File: setup/_metapackage/VERSION.txt
Comment (aisopuro) [reply setup/_metapackage/VERSION.txt]: @mlaitinen it looks like your review is maybe the only thing holding this back.

**Review by mlaitinen — APPROVED** [review]

**Pull request merged**
